.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,150 @@
+name: Regression Test
+
+on:
+  workflow_dispatch:
+
+jobs:
+  run-tests:
+    runs-on: [ubuntu-latest]
+    timeout-minutes: 20
+    if: ${{ github.event_name != 'issue_comment' || needs.check-comment-trigger.outputs.is-comment-triggered == 'true' }}
+    strategy:
+      matrix:
+        # These are all the Python versions that we want to regression test with.
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+
+    name: Run Regressions Tests with Python Version ${{ matrix.python-version }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Create the logs directory
+        run: mkdir -p logs
+
+      - name: Set filename for the server's output logs
+        run: echo "SERVER_LOGS_FILE=logs/server_logs_${{ matrix.python-version }}" >> $GITHUB_ENV
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Set up GOPATH variable
+        run: echo "GOPATH=$(echo $HOME)" >> $GITHUB_ENV
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16.3
+
+      - name: Install aqueduct-ml
+        run: pip3 install aqueduct-ml
+
+      - name: Start the server
+        run: (aqueduct start > $SERVER_LOGS_FILE 2>&1 &)
+
+      - name: Install Packages needed by the notebooks
+        run: pip3 install sklearn transformers torch
+
+      - name: Wait for server
+        timeout-minutes: 1
+        run: while ! echo exit | nc localhost 8080; do sleep 1; done
```

Comment:
Not necessary for this PR but there sure starting to be a lot of duplication for this server-setup process. Maybe we should file a task for refactoring that functionality out?

Reply:
This flow seems a bit more unique though